### PR TITLE
feat: Use dockerhub image for faster execution

### PR DESCRIPTION
**Pull request by simplesagar**
https://hub.docker.com/r/speakeasysagar/sdk-generation-action/tags

### CODE REVIEW

**Review by TristanSpeakEasy — CHANGES_REQUESTED** [review]
It is actually this line you need to change to point at your docker image https://github.com/speakeasy-api/sdk-generation-action/blob/401736c27b69aa58282f85440efd83bd06234352/action.yml#L84 and then update all the version references around the repo (should be 3 or 4 that need to be bumped to v8)

**Comment by simplesagar** [comment]
Aah right we need to update the action metadata, thanks for the spot.

File: .github/workflows/sdk-generation.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
       commit_hash: ${{ steps.generate.outputs.commit_hash }}
     steps:
       - id: generate
-        uses: speakeasy-api/sdk-generation-action@v7
+        uses: speakeasysagar/sdk-generation-action@v8
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-generation.yaml:96]:
```suggestion
        uses: speakeasy-api/sdk-generation-action@v8
```

File: action.yml [hunk 2]
```diff
@@ -81,7 +81,7 @@ outputs:
     description: "The commit hash if the SDK was regenerated"
 runs:
   using: "docker"
-  image: "Dockerfile"
+  image: "docker://speakeasysagar/sdk-generation-action:v8"
```
Comment (TristanSpeakEasy) [inline action.yml:84]: I'm not 100% sure this is how to define this here, did you find documentation on it? I'm just looking now to validate but from memory it should just be `speakeasysagar/sdk-generation-action`

Comment (simplesagar) [reply action.yml:84]: Here is the metadata syntax documentation: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsuses

```
runs:
  using: "composite"
  steps:
...
...
    # References a docker public registry action
    - uses: docker://gcr.io/cloud-builders/gradle
    # Reference a docker image published on docker hub
    - uses: docker://alpine:3.8
```

Comment (TristanSpeakEasy) [reply action.yml:84]: ah cool 👍

**Review by TristanSpeakEasy — APPROVED** [review]

**Pull request merged**